tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -156,13 +155,24 @@ def get_resource_owner_emails_from_resource(self, sqs_message, resource):
         resource_owner_tag_keys = self.config.get('contact_tags', [])
         resource_owner_tag_values = get_resource_tag_targets(resource, resource_owner_tag_keys)
         explicit_emails = self.get_valid_emails_from_list(resource_owner_tag_values)
-
-        # resolve the contact info from ldap
-        non_email_ids = list(set(resource_owner_tag_values).difference(explicit_emails))
-        ldap_emails = list(chain.from_iterable([self.ldap_lookup.get_email_to_addrs_from_uid
-                                              (uid) for uid in non_email_ids]))
-
-        return list(chain(explicit_emails, ldap_emails))
+        if explicit_emails:
+            if is_email(explicit_emails):
+                return [explicit_emails]
+            # if the LDAP config is set, lookup in ldap
+            elif self.config.get('ldap_uri', False):
+                return self.ldap_lookup.get_email_to_addrs_from_uid(explicit_emails)
+            # the org_domain setting is configured, append the org_domain
+            # to the username from AWS
+            elif self.config.get('org_domain', False):
+                org_domain = self.config.get('org_domain', False)
+                self.logger.info('adding email %s to targets.', explicit_emails + '@' + org_domain)
+                return [explicit_emails + '@' + org_domain]
```

Comment:
explicit_emails is a list of addresses, and the code in the pr is treating it as string